cognite/client/utils/_time.py

```diff
@@ -62,15 +62,19 @@ def get_utc_zoneinfo() -> ZoneInfo:
 
 
 def datetime_to_ms(dt: datetime) -> int:
-    """Converts datetime object to milliseconds since epoch.
+    """Converts a datetime object to milliseconds since epoch.
 
     Args:
         dt (datetime): Naive or aware datetime object. Naive datetimes are interpreted as local time.
 
     Returns:
         ms: Milliseconds since epoch (negative for time prior to 1970-01-01)
     """
-    return int(1000 * dt.timestamp())
+    try:
+        return int(1000 * dt.timestamp())
+    except OSError:
+        # OSError is raised if dt.timestamp() is called before 1970-01-01 on Windows for naive datetime.
+        return int(1000 * (dt - datetime(1970, 1, 1)).total_seconds())
```

Comment:
This implementation does actually raise an error. 
```python
datetime(1969, 1, 1).astimezone(tz=timezone.utc)
Traceback (most recent call last):
  File "C:\Program Files\JetBrains\PyCharm 2022.3.2\plugins\python\helpers\pydev\pydevconsole.py", line 364, in runcode
    coro = func()
  File "<input>", line 1, in <module>
OSError: [Errno 22] Invalid argument
```

Reply:
Furthermore, I am a bit skeptical of doing an implementation "just in case", I rather have it tested for the cases we are interested in.

Reply:
> This implementation does actually raise an error.

Dammit Windows 😂 

> Furthermore, I am a bit skeptical of doing an implementation "just in case", I rather have it tested for the cases we are interested in.

...but your original code does datetime subtraction under the assumption that the only way this error triggers, is when "windows + naive". I think it is better to assume we don't know when an error might arise from e,g, an "exotic edge-device OS", and use code that uses the correct strategy for computing time spans (i.e. convert to utc first).

With that said, since `dt.astimezone(timezone.utc)` also fails on Windows (me -> 🤦‍♂️ x 1000), that strategy is out of the question, I vote to keep your existing solution. I would prefer if you added a check to ensure the timestamp is naive though and if not, reraise

Reply:
What about the ugly double try except?

Reply:
added it now

Reply:
Adressed in another thread